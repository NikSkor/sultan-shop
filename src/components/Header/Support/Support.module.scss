.supportBlock {
  display: flex;
  gap: 14px;
}

.info {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.tel {
  font-weight: 600;
  font-size: 14px;
  line-height: 150%;
  // color: #111111;
}

.workTime {
  font-weight: 300;
  font-size: 12px;
  line-height: 150%;
  // color: inherit;
}

.callBack {
  font-weight: 400;
  font-size: 10px;
  line-height: 150%;
  text-decoration-line: underline;
  // color: #3F4E65;
}