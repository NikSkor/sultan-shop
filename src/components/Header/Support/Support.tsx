import React, { FC } from 'react';
import style from './Support.module.scss';

interface SupportProps {
  children?: React.ReactNode,
  telColor: string,
  textColor: string,
  alItems : string,
}

const Support: FC<SupportProps> = ({children, telColor, textColor, alItems}) => {
  return (
    <div className={style.supportBlock}>
      <div className={style.info} style={{alignItems: alItems}}>
        <a href="tel:+77774900091" className={style.tel} style={{color: telColor}}>+7 (777) 490-00-91</a>
        <p className={style.workTime} style={{color: textColor}}>время работы: 9:00-20:00</p>
        <button className={style.callBack} style={{color: textColor}}>Заказать звонок</button>
      </div>
      {children}
    </div>
  )
}

export default Support;