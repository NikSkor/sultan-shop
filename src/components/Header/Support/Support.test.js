import { render, screen } from '@testing-library/react';
import Support from './Support';

describe('test of support component', ()=> {
  test('telephone', () => {
    render(<Support />);
    const tel = screen.getByText('+7 (777) 490-00-91');
    expect(tel).toBeInTheDocument();
    expect(tel).toMatchSnapshot();
  });

  test('work time', () => {
    render(<Support />);
    const workTime = screen.getByText(/время работы: 9:00-20:00/i);
    expect(workTime).toBeInTheDocument();
    expect(workTime).toMatchSnapshot();
  })

  test('button call back', () => {
    render(<Support />);
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  })
});