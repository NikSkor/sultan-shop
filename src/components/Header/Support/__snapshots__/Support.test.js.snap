// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test of support component button call back 1`] = `
<button
  class="callBack"
>
  Заказать звонок
</button>
`;

exports[`test of support component telephone 1`] = `
<a
  class="tel"
  href="tel:+77774900091"
>
  +7 (777) 490-00-91
</a>
`;

exports[`test of support component work time 1`] = `
<p
  class="workTime"
>
  время работы: 9:00-20:00
</p>
`;
